pytorch_lightning/accelerators/gpu.py

```diff
@@ -53,6 +64,58 @@ def set_nvidia_flags(local_rank: int) -> None:
         devices = os.getenv("CUDA_VISIBLE_DEVICES", all_gpu_ids)
         _log.info(f"LOCAL_RANK: {local_rank} - CUDA_VISIBLE_DEVICES: [{devices}]")
 
+    def get_device_stats(self, device: Optional[torch.device] = None) -> Dict[str, Any]:
+        """Gets stats for the given GPU device."""
+        if _TORCH_GREATER_EQUAL_1_8:
+            return torch.cuda.memory_stats(device=device)
+        else:
+            gpu_stat_keys = [
+                ("utilization.gpu", "%"),
+                ("memory.used", "MB"),
+                ("memory.free", "MB"),
+                ("utilization.memory", "%"),
+                ("fan.speed", "%"),
+                ("temperature.gpu", "°C"),
+                ("temperature.memory", "°C"),
+            ]
+            gpu_stats = self._get_gpu_stats([k for k, _ in gpu_stat_keys])
+            device_stats = self._parse_gpu_stats(self._device_ids, gpu_stats, gpu_stat_keys)
+            return device_stats
+
+    def _get_gpu_stats(self, queries: List[str]) -> List[List[float]]:
+        if not queries:
+            return []
+
+        """Run nvidia-smi to get the gpu stats"""
+        gpu_query = ",".join(queries)
+        format = "csv,nounits,noheader"
+        gpu_ids = ",".join(self._gpu_ids)
+        result = subprocess.run(
+            [shutil.which("nvidia-smi"), f"--query-gpu={gpu_query}", f"--format={format}", f"--id={gpu_ids}"],
+            encoding="utf-8",
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,  # for backward compatibility with python version 3.6
+            check=True,
+        )
+
+        def _to_float(x: str) -> float:
+            try:
+                return float(x)
+            except ValueError:
+                return 0.0
+
+        stats = result.stdout.strip().split(os.linesep)
+        stats = [[_to_float(x) for x in s.split(", ")] for s in stats]
+        return stats
+
+    @staticmethod
+    def _get_gpu_ids(device_ids: List[int]) -> List[str]:
+        """Get the unmasked real GPU IDs."""
+        # All devices if `CUDA_VISIBLE_DEVICES` unset
+        default = ",".join(str(i) for i in range(torch.cuda.device_count()))
+        cuda_visible_devices: List[str] = os.getenv("CUDA_VISIBLE_DEVICES", default=default).split(",")
+        return [cuda_visible_devices[device_id].strip() for device_id in device_ids]
+
```

Comment:
let's call the memory utils here: https://github.com/PyTorchLightning/pytorch-lightning/blob/c7451b3ccf742b0e8971332caf2e041ceabd9fe8/pytorch_lightning/utilities/memory.py#L126-L152

Reply:
`get_gpu_memory_map` only queries for `memory.used`, not the other fields like:
```
("utilization.gpu", "%"),
                ("memory.used", "MB"),
                ("memory.free", "MB"),
                ("utilization.memory", "%"),
                ("fan.speed", "%"),
                ("temperature.gpu", "°C"),
                ("temperature.memory", "°C"),
```

are you saying we only need `memory.used`?

Reply:
The function `_get_gpu_stats` was taken from gpu_stats_monitor. I was actually thinking we could deprecate `get_gpu_memory_map` since it is almost the same as `_get_gpu_stats`

Reply:
I see, I do think we ought to merge the utilities somewhere and not have them spread across the codebase like this. @awaelchli @carmocca do you think having a `utilities/gpu.py` makes sense to capture this? then the GPU accelerator simply calls the utils function there.

Reply:
I updated the PR with a utility function `_get_nvidia_gpu_stats`. Personally I think it makes sense to just keep it at the bottom of gpu.py, since it's a short file and I don't think the func will be used in other places, but can also move it elsewhere.

My plan is to mark `get_gpu_memory_map` and `get_memory_profile` as deprecated at the same as marking `log_gpu_memory` as deprecated, and then remove them all in 1.7.